app/src/main/java/io/intrepid/contest/screens/entrysubmission/join/JoinContract.java

```diff
@@ -2,18 +2,22 @@
 
 import io.intrepid.contest.base.BaseContract;
 
-public class JoinContract {
+class JoinContract {
     interface View extends BaseContract.View {
         void showSubmitButton();
 
         void hideSubmitButton();
 
         void showEntryNameScreen();
+
+        void showInvalidCodeErrorMessage();
```

Comment:
Good point! But the message itself is a string resource so it belongs to the view; and I think it would add complexity creating message types in this case that we only have two that are very different.